src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -1630,4 +1686,17 @@ private void logAttackMoves(final List<ProTerritory> prioritizedTerritories) {
       }
     }
   }
+
+  private boolean canAirSafelyLandAfterAttack(final Unit unit, final Territory t) {
+    final boolean isAdjacentToAlliedFactory =
+        Matches.territoryHasNeighborMatching(data,
+            ProMatches.territoryHasInfraFactoryAndIsAlliedLand(player, data)).match(t);
+    final int range = TripleAUnit.get(unit).getMovementLeft();
+    final int distance =
+        data.getMap().getDistance_IgnoreEndForCondition(ProData.unitTerritoryMap.get(unit), t,
+            ProMatches.territoryCanMoveAirUnitsAndNoAA(player, data, true));
+    final boolean usesMoreThanHalfOfRange = distance > range / 2;
+    return isAdjacentToAlliedFactory || !usesMoreThanHalfOfRange;
```

Comment:
I disagree. Reading this method, it akes time to udnerstand every line up to 1699. Then you get to learn that you wasted time reading figuring out 3 lines of code, and wasted computation as it had no effect. It'd be better to have multiple return points than to have unnecessary execution. The simpler helper methods would make it all simple, which I think is more ideal